examples/pxScene2d/src/pxFileCache.cpp

```diff
@@ -0,0 +1,349 @@
+    return RT_ERROR;
+  }
+  return RT_OK;
+}
+
+rtError rtFileCache::addToCache(const rtHttpCacheData& data)
+{
+  rtString url;
+  rtError  err = data.url(url);
+  if (RT_OK != err)
+    return err;
+
+  if  (url.isEmpty())
+    return RT_ERROR;
+
+  rtString filename =  getHashedFileName(url);
+  if (filename.isEmpty())
+  {
+    rtLogWarn("Problem in getting hash from the url(%s) while adding to cache ",url.cString());
+    return RT_ERROR;
+  }
+
+  bool ret = writeFile(filename,data);
+  if (true != ret)
+     return RT_ERROR;
+  setFileSizeAndTime(filename);
+  mCurrentSize += mFileSizeMap[filename];
+  int64_t size = cleanup();
+  rtLogWarn("current size after insertion and cleanup (%d)",size);
+  return RT_OK;
+}
+
+rtError rtFileCache::getHttpCacheData(const char* url, rtHttpCacheData& cacheData)
+{
+  rtString urlToQuery = url;
+  rtString filename =  getHashedFileName(urlToQuery);
+  if (filename.isEmpty())
+  {
+    rtLogWarn("Problem in getting hash from the url(%s) while read from cache",url);
+    return RT_ERROR;
+  }
+  if (false == readFileHeader(filename,cacheData))
+    return RT_ERROR; 
+  return RT_OK;
+}
+
+void rtFileCache::clearCache() 
+{
+  string cmd = "rm -rf ";
+  cmd.append(mDirectory.cString());
+  cmd.append("/*");
+  system(cmd.c_str());
+  mFileSizeMap.clear();
+  mCurrentSize = 0;
+}
+
+int64_t rtFileCache::cleanup()
+{
+  if ( (mCurrentSize > mMaxSize) && !(mFileTimeMap.empty()))
+  {
+    map<time_t,rtString>::iterator iter = mFileTimeMap.begin();
+    vector <time_t> timeMapIters;
+    do
+    {
+      rtString filename = iter->second;
+      if (! filename.isEmpty())
+      {
+          if(false == deleteFile(filename))
+          {
+            rtLogWarn("!!! deletion of cache failed during cleanup for file(%s)",filename.cString());
+          }
+          else
+          {
+            mCurrentSize = mCurrentSize - mFileSizeMap[filename];
+            timeMapIters.push_back(iter->first);
+            mFileSizeMap.erase(filename);
+          }
+      }
+      iter++;
+    } while ((mCurrentSize > mMaxSize) && (iter != mFileTimeMap.end()));
+
+    for (int count =0; count < timeMapIters.size(); count++)
+      mFileTimeMap.erase(timeMapIters[count]);
+    timeMapIters.clear();
+  }
+  return mCurrentSize;
+}
+
+rtString rtFileCache::getHashedFileName(const rtString& url)
+{
+  long int hash = hashFn(url.cString());
+  stringstream stream;
+  stream << hash;
+  return stream.str().c_str();
+}
+
+void rtFileCache::setFileSizeAndTime(rtString& filename)
+{
+  if (!mDirectory.isEmpty())
+  {
+    struct stat statbuf;
+    rtString absPath  = getAbsPath(filename);
+    if (stat(absPath.cString(), &statbuf) == 0)
+    {
+      mFileSizeMap[filename] = statbuf.st_size;
+      mFileTimeMap[statbuf.st_atim.tv_sec] = filename;
+    }
+  }
+}
+
+bool rtFileCache::writeFile(rtString& filename,const rtHttpCacheData& cacheData)
+{
+  rtData data;
+  string date = to_string(cacheData.expirationDateUnix());
```

Comment:
to_string() is not supported on all platforms.  Please change to:

Change from to_string() to the below to support all platforms:

stringstream stream;
stream << cacheData.expirationDateUnix();
string date = stream.str().c_str();